dev/Tiles/entity.h

```diff
@@ -0,0 +1,66 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include "metatile.h"
+
+typedef struct Entity {
+    MetaTile* metatile[32];
+    unsigned char frameRate;
+    unsigned char numFrames;
+    unsigned int currentFrame;
+} Entity;
+
+Entity* Entity_Create(MetaTile* metatile[], unsigned char numFrames) 
+{
+    Entity* entity = malloc(sizeof(struct Entity));
+    // entity->metatile = malloc(sizeof(struct MetaTile) * 32);
+    entity->currentFrame = 0;
+    entity->frameRate = 30;
+    entity->numFrames = numFrames;
+
+
+    unsigned char i;
+    for(i = 0; i < metatile[i]->numTiles; ++i)
+    {
+        entity->metatile[i] = metatile[i];
+    }
+
+    return entity;
+}
+
+void Entity_Delete(Entity* entity)
+{
+    if(entity != NULL)
+        free(entity);
+}
+
+void Entity_Update(Entity* entity, unsigned int time)
+{
+    if((time % entity->frameRate) == 0)
+    {
+        entity->currentFrame++;
+    }
+    
+    if(entity->currentFrame > entity->numFrames)
+    {
+        entity->currentFrame = 0;
+    }
+}
+
+unsigned char Entity_GetCurrentFrame(Entity* entity)
```

Comment:
It's a little unusual to write getters in C I'd say, mostly because there aren't any access modifiers in any case. For example, your caller has access to all the fields in `entity` anyway, so why do they need you to pull the value out and return it? It adds some ambiguity about the "correct" way to retrieve the value of current frame - via the accessor or directly via the struct.

Also, I'm not sure if this is intentional or not, but you are returning `unsigned char` here whereas your `currentFrame` field looks to be an `unsigned int`. I would have expected the compiler to warn about this.

Reply:
FrameRate should actually be an unsigned char - It's actually a modulo against a current_frame update in main.c - Will change that back as I think it was left in during debugging. Thanks

Reply:
Re: getter. Yeah, this was to act as a hint to what would be a public in any other language as I would also intent to add a setter. Private / protected can't be enforced but it felt like exposing things this way would help out the user as I could add documentation code above these functions and that would be able to then be built out as auto docs.